### PR TITLE
new package: miniGMG

### PR DESCRIPTION
**Pull request by jiseung**

### CODE REVIEW

File: var/spack/repos/builtin/packages/minigmg/package.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    version('cuda', url='http://crd.lbl.gov/assets/Uploads/FTG/Projects/miniGMG/miniGMG.cuda.tar.gz')
+
+    depends_on('mpi')
+    depends_on('cuda', when='@cuda')
+
+    phases = ['build', 'install']
+
+    def build(self, spec, prefix):
+        cc = Executable(spec['mpi'].mpicc)
+        cc('-O3', '{0}'.format(self.compiler.openmp_flag), 'miniGMG.c',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/minigmg/package.py]: You don't need to convert `self.compiler.openmp_flag` to a string, it already is a string.

File: var/spack/repos/builtin/packages/minigmg/package.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            'mg.c', 'box.c', 'solver.c', 'operators.ompif.c', 'timer.x86.c',
+            '-D__MPI', '-D__COLLABORATIVE_THREADING=6',
+            '-D__TEST_MG_CONVERGENCE', '-D__PRINT_NORM', '-D__USE_BICGSTAB',
+            '-o', 'run.miniGMG', '-lm')
+
+    def install(self, spec, prefix):
+        mkdir(prefix.bin)
+        install('run.miniGMG', prefix.bin)
+        mkdir(prefix.jobs)
+        files = glob.glob(join_path('./', 'job*'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/minigmg/package.py]: I don't think you need `./` here, glob assumes a relative path.

File: var/spack/repos/builtin/packages/minigmg/package.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    and algorithmic tradeoffs inherit in multigrid. miniGMG was developed
+    under the CACHE Joint Math-CS Institute.
+
+    Note, miniGMG code has been supersceded by HPGMG. """
+
+    homepage = "http://crd.lbl.gov/departments/computer-science/PAR/research/previous-projects/miniGMG/"
+    url      = "http://crd.lbl.gov/assets/Uploads/FTG/Projects/miniGMG/miniGMG.tar.gz"
+
+    version('master', '975a2a118403fc0024b5e04cef280e95')
+    version('cuda', url='http://crd.lbl.gov/assets/Uploads/FTG/Projects/miniGMG/miniGMG.cuda.tar.gz')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/minigmg/package.py]: This version is missing a checksum.

**Comment by jiseung** [comment]
@adamjstewart There aren't any instructions available on how to build with the cuda version, and I couldn't guess how to do it. I'm hoping to add the version once I get more information from the developers.

**Pull request merged**